### PR TITLE
Added the object representation of the CloudWatch Alarm

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*

https://github.com/aws/aws-lambda-java-libs/issues/484

*Description of changes:*

I added a test dependency to serialisation so I could write a unit test with the correct serialisation configuration for Jackson.

*Target (OCI, Managed Runtime, both):*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by msailes** [comment]
There is a design decision on whether there should be one class or two? The two different types of alarms use different attributes.

**Review by smirnoal — APPROVED** [review]

File: aws-lambda-java-events/pom.xml [hunk 1]
```diff
@@ -69,6 +71,18 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.skyscreamer</groupId>
+            <artifactId>jsonassert</artifactId>
+            <version>1.5.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-lambda-java-serialization</artifactId>
```
Comment (smirnoal) [inline aws-lambda-java-events/pom.xml:82]: Can you move the test to `aws-lambda-java-tests` instead?

File: aws-lambda-java-events/src/test/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEventTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.skyscreamer.jsonassert.JSONAssert.assertEquals;
+import static org.skyscreamer.jsonassert.JSONCompareMode.STRICT;
+
+public class CloudWatchAlarmEventTest {
```
Comment (maxday) [inline aws-lambda-java-events/src/test/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEventTest.java:18]: It seems that those tests are not properly run in the CI? See GitHub action output:
```
2024-06-14T21:14:00.1302461Z Running com.amazonaws.services.lambda.runtime.events.CloudWatchAlarmEventTest
2024-06-14T21:14:00.1303887Z Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
```

**Comment by msailes** [comment]
What do you think about having both use cases model as a single class?

I think having two is awkward because some attributes won't make sense in one case. But the alternative of having two is awkward if CloudWatch continues to make different events in the future.

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEvent.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+import java.util.Map;
+
+@Data
+@Builder(setterPrefix = "with")
+@NoArgsConstructor
+@AllArgsConstructor
+public class CloudWatchAlarmEvent {
```
Comment (maxday) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEvent.java:15]: let's add the new event here -> https://github.com/aws/aws-lambda-java-libs/blob/main/aws-lambda-java-events/README.md

File: aws-lambda-java-events/pom.xml (hunk 1 above)
Comment (msailes) [reply aws-lambda-java-events/pom.xml]: Done

File: aws-lambda-java-events/src/test/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEventTest.java (hunk 2 above)
Comment (msailes) [reply aws-lambda-java-events/src/test/java/com/amazonaws/services/lambda/runtime/events/CloudWatchAlarmEventTest.java]: Resolved

**Comment by msailes** [comment]
Ready to recheck

**Review by smirnoal — APPROVED** [review]

File: aws-lambda-java-events/pom.xml [hunk 4]
```diff
@@ -35,6 +35,8 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <lombok.version>1.18.22</lombok.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
```
Comment (maxday) [inline aws-lambda-java-events/pom.xml:39]: @msailes is this change required in this PR?

**Review by maxday — APPROVED** [review]
